File: app/src/main/java/com/mdmobile/cyclops/ui/logIn/LoginConfigureServerFragment.java
package com.mdmobile.cyclops.ui.logIn;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;
import com.mdmobile.cyclops.R;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import static com.mdmobile.cyclops.utils.GeneralUtility.validateUrl;


public class LoginConfigureServerFragment extends Fragment implements View.OnFocusChangeListener {

    private EditText serverAddressEditText, serverNameEditText;

    public LoginConfigureServerFragment() {
        //Empty constructor required
    }

    public static LoginConfigureServerFragment newInstance() {
        return new LoginConfigureServerFragment();
    }

    //Interfaces
    @Override
    public void onFocusChange(View v, boolean hasFocus) {
        if (hasFocus) {
            return;
        }
        switch (v.getId()) {
            case R.id.instance_name_text_view:
                ((LoginActivity) getActivity()).viewModel.updateInstanceName(((EditText) v).getText().toString());
                break;
            case R.id.instance_address_text_view:
                ((LoginActivity) getActivity()).viewModel.updateInstanceAddress(((EditText) v).getText().toString());
                break;
            default:
                throw new IllegalArgumentException("Unknown view ID: " + v.getId());
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_server_name, container, false);
        serverAddressEditText = rootView.findViewById(R.id.instance_address_text_view);
        serverNameEditText = rootView.findViewById(R.id.instance_name_text_view);

        serverNameEditText.setOnFocusChangeListener(this);
        serverAddressEditText.setOnFocusChangeListener(this);

        return rootView;
    }

    public void serverConnectionTest(View v) {
        RequestQueue requestQueue = Volley.newRequestQueue(getContext());
        String serverUrl = serverAddressEditText.getText().toString();

        serverUrl = validateUrl(serverUrl);

        StringRequest stringRequest = new StringRequest(Request.Method.GET, serverUrl,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {

                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });
    }
}

File: app/src/main/java/com/mdmobile/cyclops/ui/logIn/LoginConfigureServerFragment.kt
package com.mdmobile.cyclops.ui.logIn

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.mdmobile.cyclops.R
import kotlinx.android.synthetic.main.fragment_server_name.*


class LoginConfigureServerFragment : Fragment(), View.OnFocusChangeListener {
    companion object {
        fun newInstance(): LoginConfigureServerFragment {
            return LoginConfigureServerFragment()
        }
    }

    //Interfaces
    override fun onFocusChange(v: View, hasFocus: Boolean) {
        if (hasFocus) {
            return
        }
        val text = (v as EditText).text.toString()
        when (v.id) {
            R.id.instance_name_edit_text -> (activity as LoginActivity).viewModel.updateInstanceName(text)
            R.id.instance_address_edit_text -> (activity as LoginActivity).viewModel.updateInstanceAddress(text)
            else -> throw IllegalArgumentException("Unknown view ID: " + v.id)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_server_name, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        instance_address_edit_text.onFocusChangeListener = this
        instance_name_edit_text.onFocusChangeListener = this
    }

//    fun serverConnectionTest(v: View) {
//        val requestQueue = Volley.newRequestQueue(context)
//        var serverUrl = serverAddressEditText!!.text.toString()
//
//        serverUrl = validateUrl(serverUrl)
//
//        val stringRequest = StringRequest(Request.Method.GET, serverUrl,
//                Response.Listener { }, Response.ErrorListener { })
//    }

}
